### ISSUE TITLE
Warning 32: unused value string_of__1

### ISSUE THREAD

**Issue by rauanmayemir** [mjambon/atd#30]
Those warnings can even be found in [RWO v1](https://realworldocaml.org/v1/en/html/handling-json-data.html). If I run `jbuilder` with `--dev` making it use stricter compilation flags by default, I get multiple warnings like:

```
File "src/types_j.ml", line 25, characters 4-16:
Warning 32: unused value string_of__1.
File "src/types_j.ml", line 597, characters 4-16:
Warning 32: unused value string_of__2.
File "src/types_j.ml", line 613, characters 4-16:
Warning 32: unused value string_of__3.
File "src/types_j.ml", line 1:
Error: Some fatal warnings were triggered (3 occurrences)
```

I added an example to reproduce the issue here: [https://github.com/rauanmayemir/atdgen-example](https://github.com/rauanmayemir/atdgen-example)

CC: @rgrinberg

**Comment by mjambon** [comment]
Thanks for reporting this.

What's going on here is that we generate the `string_of_XXX` functions even though they're not exported. I think they're never used internally (needs to be checked) so we could simply not generate them. It seems like a feasible task for an adventurous hacker who wants to get it done soon, so I'm marking this bug as such. I don't have the resources to tackle this myself at the moment.

Now, it's possible that other unused functions pop up since it's not something I worried about when writing the code generator. A workaround for the user is to turn off warning 32 when compiling the ml files produced by atdgen.

**Comment by rgrinberg** [comment]
I made a fix in #31 . It's not the best fix, because we'd really prefer not to generate code that ends up unused. But there are other warnings causing issues and this isn't so trivial to fix. So I just made the generators disable warnings in their generated code.
